tests/drift/test_drift.py

```diff
@@ -584,3 +596,81 @@ def test_result_comparison_to_cbpe_plots_raise_no_exceptions(sample_drift_data):
+    assert result.categorical_column_names == ['f3', 'f4']
+    assert result.timestamp_column_name == 'timestamp'
+    assert result.continuous_method_names == ['kolmogorov_smirnov']
+    assert result.categorical_method_names == ['chi2']
+    assert [m.column_name for m in result.continuous_methods] == ['kolmogorov_smirnov']
+    assert [m.column_name for m in result.categorical_methods] == ['chi2']
+    assert sorted(m.column_name for m in result.methods) == sorted(('chi2', 'kolmogorov_smirnov'))
+
+
+def test_univariate_drift_result_filter_should_preserve_data_with_default_args(univariate_drift_result):
+    filtered_result = univariate_drift_result.filter()
+    assert filtered_result.data.equals(univariate_drift_result.data)
+    assert filtered_result.continuous_column_names == univariate_drift_result.continuous_column_names
+    assert filtered_result.categorical_column_names == univariate_drift_result.categorical_column_names
+    assert filtered_result.timestamp_column_name == univariate_drift_result.timestamp_column_name
+    assert filtered_result.continuous_method_names == univariate_drift_result.continuous_method_names
+    assert filtered_result.categorical_method_names == univariate_drift_result.categorical_method_names
+    assert filtered_result.continuous_methods == univariate_drift_result.continuous_methods
+    assert filtered_result.categorical_methods == univariate_drift_result.categorical_methods
+
+
+def test_unvariate_drift_result_filter_metrics(univariate_drift_result):
+    filtered_result = univariate_drift_result.filter(methods=['chi2'])
+    metrics = tuple(set(metric for (_, metric, _) in filtered_result.data.columns if metric != 'chunk'))
+    assert metrics == ('chi2',)
+    assert filtered_result.data.shape[0] == univariate_drift_result.data.shape[0]
+
+    assert filtered_result.continuous_column_names == ['f1', 'f2']
+    assert filtered_result.categorical_column_names == ['f3', 'f4']
+    assert filtered_result.continuous_method_names == []
+    assert filtered_result.categorical_method_names == ['chi2']
+    assert [m.column_name for m in filtered_result.continuous_methods] == []
+    assert [m.column_name for m in filtered_result.categorical_methods] == ['chi2']
+    assert sorted(m.column_name for m in filtered_result.methods) == ['chi2']
+
+
+def test_unvariate_drift_result_filter_column_names(univariate_drift_result):
+    filtered_result = univariate_drift_result.filter(column_names=['f1', 'f2'])
+    columns = tuple(sorted(set(column for (column, _, _) in filtered_result.data.columns if column != 'chunk')))
+    assert columns == ('f1', 'f2')
+    assert filtered_result.data.shape[0] == univariate_drift_result.data.shape[0]
+
+    assert filtered_result.continuous_column_names == ['f1', 'f2']
+    assert filtered_result.categorical_column_names == []
+    assert filtered_result.continuous_method_names == ['kolmogorov_smirnov']
+    assert filtered_result.categorical_method_names == ['chi2']
```

Comment:
Same idea as above, this seems like a bug where the categorical method names are retained even though the columns have been filtered.